### PR TITLE
Clearer naming of nat override options

### LINKED ISSUE #874: Rename WithReachability option
This option _did not_ pass the "wtf is this thing" test with @daviddias. We should consider calling it something like `ForceReachability` or even `AssumePublic`/`AssumePrivate`, ... I'm not sure. Just something else.

Thoughts @willscott?

### PR DESCRIPTION
**Pull request by willscott**
fix #874

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -279,15 +279,20 @@ func DefaultStaticRelays() Option {
 	}
 }
 
-// WithReachability overrides automatic reachability detection to force the local node
-// to believe it is either unreachable or reachable externally.
-func WithReachability(reachable bool) Option {
+// WithNATPublicReachability overrides automatic reachability detection to force the local node
+// to believe it is reachable externally.
+func WithNATPublicReachability() Option {
```
Comment (Stebalien) [inline options.go:284]: * NAT isn't _actually_ relevant. AutoNAT is poorly named...
* `With` is also funky given that we're not using that idiom for other libp2p options.
* Given the description, I fell like this should use the word force somewhere.

Maybe:

* `ForceReachabilityPublic`, `ForceReachabilityPrivate`?

**Comment by Stebalien** [comment]
@daviddias better?

**Comment by daviddias** [comment]
Not 100% sure, will sleep on it. I think the thing that makes it look foreign is that even for a Networks person, "reachability" as a property is not directly associated with NAT'ed vs. Public IPs. So it is a stepping stone to learn about, not huge though and the docs do help :)

**Comment by Stebalien** [comment]
> I think the thing that makes it look foreign is that even for a Networks person, "reachability" as a property is not directly associated with NAT'ed vs. Public IPs.

That's actually what this option is trying to capture. The "AutoNAT" service is really trying to determine "am I directly reachable from the public internet", not "am I behind a NAT".

**Pull request merged**